datahub-actions/src/datahub_actions/plugin/action/execution/executor_action.py

```diff
@@ -174,31 +174,36 @@ def _handle_execution_request_signal(self, orig_event):
     def _build_executor_config(
         self, config: ExecutorConfig, ctx: PipelineContext
     ) -> ReportingExecutorConfig:
+        if not config.task_configs:
+            # Build default task config
+            ingest_task_config = TaskConfig(
+                name="RUN_INGEST",
+                type="acryl.executor.execution.sub_process_ingestion_task.SubProcessIngestionTask",
+                configs=dict({}),
+            )
 
-        # Build default task config
-        ingest_task_config = config.task_config or TaskConfig(
-            name="RUN_INGEST",
-            type="acryl.executor.execution.sub_process_ingestion_task.SubProcessIngestionTask",
-            configs=dict({}),
-        )
-
-        test_connection_task_config = TaskConfig(
-            name="TEST_CONNECTION",
-            type="acryl.executor.execution.sub_process_test_connection_task.SubProcessTestConnectionTask",
-            configs={},
-        )
+            test_connection_task_config = TaskConfig(
+                name="TEST_CONNECTION",
+                type="acryl.executor.execution.sub_process_test_connection_task.SubProcessTestConnectionTask",
+                configs={},
+            )
 
         if not ctx.graph:
             raise Exception(
                 "Invalid configuration provided to action. DataHub Graph Client Required. Try including the 'datahub' block in your configuration."
             )
 
+        task_configs = config.task_configs or [
```

Comment:
Can we move this into an `else` that corresponds with the above `if`?

Just want to co-locate related logic in our code

Reply:
I have changed it a bitm should be better now. 😊